coordinator/client/zookeeper/connection.go

```diff
@@ -159,43 +152,23 @@ func (c *Connection) CreateDir(path string) error {
 }
 
 func (c *Connection) createDir(p string) error {
-	return c.create(p, &dir{})
+	pth := path.Join(c.basePath, p)
+	_, err := c.conn.Create(pth, []byte{}, 0, zklib.WorldACL(zklib.PermAll))
+	return xlateError(err)
 }
 
 func (c *Connection) ensurePath(p string) error {
 	dp := path.Dir(p)
-	if ok, err := c.exists(dp); err != nil {
+	if p == "" || p == "/" {
```

Comment:
path.Dir("/') and path.Dir("") are reflective.

To your second comment, I agree, and will fix.  Thanks!